src/GraphQLParser/ParserContext.cs

```diff
@@ -37,6 +41,23 @@ public ParserContext(ROM source, ParserOptions options)
             Advance();
         }
 
+        private void IncreaseDepth()
+        {
+            // Encourage compiler inlining of this method by moving exception to a separate method
+            if (_currentDepth++ >= _maxDepth)

```

Comment:
```suggestion
            if (++_currentDepth >= _maxDepth)

```

Please add test with max depth = 0. It should work on simple values.

Reply:
Sure I can add a test.  But why the change?

Reply:
Your test should work on literal values (depth 0) and should not work on something else (depth 1).
`a++` returns a, `++a` returns incremented a, so without suggested fix test will fail I think.

Reply:
As it is now, max depth must be 1 to work on something simple like `scalar Test`.  Max depth must be 2 to work on a basic query like `{a}`.  Seems to make sense to me.  Making the change you suggested would require MaxDepth settings of 2 and 3 respectively.  We can initialize `_currentDepth` to `-1` if we want to, essentially, add one level to the maximum depth setting.

Reply:
> Making the change you suggested would require MaxDepth settings of 2 and 3 respectively.

Tested and confirmed

Reply:
> We can initialize `_currentDepth` to `-1` if we want to, essentially, add one level to the maximum depth setting.

Please review and lmk if I should make this change, so MaxDepth = 0 will parse simple things like `scalar Test`

Reply:
It depends on what numbering we have - from 0 or from 1.

Reply:
Worth to write about it in the comments. For me there is no big difference 0 or 1 to choose as "initial ground level". It should simply be noted clearly.

Reply:
Added note "Minimum value is 1` in the xml comment.